### PR TITLE
TPC: Adding option to make timegain calib with gausian fits

### PR DESCRIPTION
**Pull request by matthias-kleiner**
- setting tighter default momentum range for filtering MIPs to reject kaons and electrons

- adding option to write CalibdEdx to local file by converting the boost histogram to a ROOT histogram

- lowering minimum dEdx to 10 for better stabillity in A16

other changes:
- TPC: make calculation of dedx parallelisable

### CODE REVIEW

File: Detectors/TPC/calibration/src/CalculatedEdx.cxx [hunk 1]
```diff
@@ -53,7 +53,7 @@ void CalculatedEdx::setRefit(const unsigned int nHbfPerTf)
   mRefit = std::make_unique<o2::gpu::GPUO2InterfaceRefit>(mClusterIndex, &mTPCCorrMapsHelper, mFieldNominalGPUBz, mTPCTrackClIdxVecInput->data(), nHbfPerTf, mTPCRefitterShMap.data(), mTPCRefitterOccMap.data(), mTPCRefitterOccMap.size());
 }
 
-void CalculatedEdx::fillMissingClusters(int missingClusters[4], float minChargeTot, float minChargeMax, int method)
+void CalculatedEdx::fillMissingClusters(int missingClusters[4], float minChargeTot, float minChargeMax, int method, std::array<std::vector<float>, 5>& chargeMaxROC, std::array<std::vector<float>, 5>& mChargeMaxROC)
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/CalculatedEdx.cxx]: I'm not sure I understand why `mChargeMaxROC` which has naming convention of a member is passed as local. Also, the naming of the passed variables `chargeMaxROC` and `mChargeMaxROC` is nearly identical.

**Review by wiechula — CHANGES_REQUESTED** [review]
Thanks @matthias-kleiner , just one comment.

File: Detectors/TPC/calibration/src/CalculatedEdx.cxx (hunk 1 above)
Comment (matthias-kleiner) [reply Detectors/TPC/calibration/src/CalculatedEdx.cxx]: I accidently forgot to change it to `chargeMaxROC` and `chargeTotROC`. I just fixed it

**Review by wiechula — APPROVED** [review]

File: Detectors/TPC/calibration/src/CalculatedEdx.cxx (hunk 1 above)
Comment (wiechula) [reply Detectors/TPC/calibration/src/CalculatedEdx.cxx]: Perfect, thanks!

**Comment by wiechula** [comment]
+aync-label async-2024-pp-apass1, async-2024-pp-cpass0, async-2024-PbPb-cpass0, async-2024-PbPb-apass1, async-2024-ppRef-apass1

**Pull request merged**